src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -293,9 +272,34 @@ public static SolutionStateChecksums Deserialize(ObjectReader reader)
                     result[projectStateChecksums.Checksum] = projectStateChecksums;
                 }
             }
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Are there cases where we don't want to just return after the loop? (ie, the asset hint led us astray)

Reply:
ok.  so good point.  I think i no longer want this to be conisdered a 'hint'.  it's a strong guarantee.  do you have a ebtter name here?

Expectation is all callers provide the right information.  Note: there are LOTS of tests for this.

Reply:
ok.  i'm going to call this the AssetPath

Reply:
In my head, something like this commit is a bit easier to understand wrt using the assetpath to limit the search.

[https://github.com/ToddGrun/roslyn/commit/cce1a5a41cf81b7a1d3d1f1ce2939b562269e7fe](https://github.com/ToddGrun/roslyn/commit/cce1a5a41cf81b7a1d3d1f1ce2939b562269e7fe)

Reply:
i'll try that out!